kernel/src/osquery.cpp

```diff
@@ -342,7 +278,7 @@ static int osquery_ioctl(dev_t dev, u_long cmd, caddr_t data,
   switch (cmd) {
     case OSQUERY_IOCTL_SUBSCRIPTION:
       sub = (osquery_subscription_args_t *)data;
-      if ((err = subscribe_to_event(sub->event, sub->subscribe))) {
+      if ((err = subscribe_to_event(sub->event, sub->subscribe, sub->udata))) {
```

Comment:
Is there any way to guarantee a size of `data`? Or at least to make sure it's not `== NULL`?